lib/content_consistency_checker.rb

```diff
@@ -3,92 +3,124 @@
 class ContentConsistencyChecker
   attr_reader :errors
 
-  def initialize(content_id, locale = "en", ignore_recent = false)
-    @content_id = content_id
-    @locale = locale
-    @ignore_recent = ignore_recent
-    @errors = []
+  def initialize(content_store, content_dump)
+    @content_store = content_store
+    @content_dump = content_dump
+    @checked_content = Set.new
+    @errors = Hash.new { |hash, key| hash[key] = [] }
   end
 
-  def call
-    unless document
-      errors << "Document (#{content_id}, #{locale}) could not be found."
-      return errors
+  def check_editions
+    editions_to_check.find_each do |edition|
+      check_edition(edition)
     end
+  end
+
+  def check_content
+    unchecked_content.each do |path|
+      content_item = content_dump.fetch(path)
+      next if content_item.content_id.nil?
+      next if content_item.gone?
+      next unless content_item.schema_name.present?
 
-    check_edition("Draft", draft, draft_content_store) if draft
-    check_edition("Live", live, live_content_store) if live
+      edition = Edition.find_by(
+        content_store: content_store,
+        base_path: content_item.base_path
+      )
+      next if edition
 
-    errors
+      errors[content_item.base_path] << "No edition available."
+    end
   end
 
 private
 
-  attr_reader :content_id, :locale, :ignore_recent
+  attr_reader :content_store, :content_dump, :checked_content
+  attr_writer :errors
 
-  def check_edition(prefix, edition, content_store)
-    return unless edition.base_path
+  def unchecked_content
+    Set.new(content_dump.keys) - checked_content
+  end
+
+  def editions_to_check
+    Edition
+      .with_document
+      .where(content_store: content_store)
+  end
+
+  def get_content_item(path)
+    begin
+      path = path.to_sym
+      content_item = content_dump.fetch(path)
+      checked_content.add?(path)
+      content_item
+    rescue KeyError
+      nil
+    end
+  end
+
+  def hash_field(object)
+    Digest::SHA1.hexdigest(
+      JSON.generate(object)
+    )
+  end
+
+  def draft?
+    content_store == "draft"
```

Comment:
defensive but you might want to run `.to_s` on this incase this gets called with a symbol.